attestation-agent/attester/src/tdx/mod.rs

```diff
@@ -81,7 +84,17 @@ impl Attester for TdxAttester {
         _register_index: Option<u64>,
     ) -> Result<()> {
         for event in events {
-            match tdx_attest_rs::tdx_att_extend(&event) {
+            let mut event_buffer = [0u8; mem::size_of::<tdx_attest_rs::tdx_rtmr_event_t>()];
+            let mut hasher = Sha384::new();
+            hasher.update(&event);
+            let hash = hasher.finalize().to_vec();
+            let rtmr_event = unsafe {
```

Comment:
Any kind of validation we can do here before the unsafe cast? This API could be called by anyone in user space, right?

Reply:
The definition of the underlying `tdx_rtmr_event_t` is [here](https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/QuoteGeneration/quote_wrapper/tdx_attest/tdx_attest.h#L89-L96)

The lib inside will [cast it too](https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/QuoteGeneration/quote_wrapper/tdx-attest-rs/src/lib.rs#L208). 
If we have an same size of memory buffer `event_buffer` as `tdx_rtmr_event_t`, we can cast the struct. What we need to do is to fill some fields as what `DCAP` expects, which is implemented in the following sentences.

I think the unavoidable `unsafe` here is because the underlying crate `tdx-attest-sys` uses [bindgen](https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/QuoteGeneration/quote_wrapper/tdx-attest-sys/build.rs#L59-L78) which will only convert a C struct into a Rust struct without making it `pub` or including any (de)serialization implementations. The result is we can only use memory and pointers to do the dirty (de)serialization work.

Reply:
btw, I've get in touch with DCAP team to update the code. Hopefully after some updates we can avoid unsafe calling in upper stack, s.t. in AA.

Reply:
@fitzthum can we merge this PR and update `unsafe` part when dcap team upgrade their API?

Reply:
Yeah that's fine. Sorry for the delay.